tests/pfc/test_pfc_pause_lossless.py

```diff
@@ -0,0 +1,352 @@
+                "Background traffic should not be impacted")
+           
+            if ((tolerance_ratio < TOLERANCE_PARAMETER) or
+                (tolerance_ratio > 1)) :
+                logger.error("Expected Tx/Rx = %s actual Rx = %s"
+                    %(exp_tx_byte, exp_tx_byte))
+
+                logger.error("tolerance_ratio = %s" %(tolerance_ratio))
+
+                pytest_assert(False, 
+                    "expected % of packets not received at the RX port")
+
+    stop_traffic(session)
+
+
+def test_pfc_pause_single_lossless_priority(testbed, 
+                                            conn_graph_facts, 
+                                            lossless_prio_dscp_map,
+                                            duthost, 
+                                            ixia_dev, 
+                                            ixia_api_server_session,
+                                            fanout_graph_facts):
+    """
+    RDMA PFC - Pauses on single lossless priority.
+    1. On SONiC DUT enable PFC on any priorities Pi. (0 <= i <= 7).
+    2. Disable the PFC watchdog on the SONiC DUT.
+    3. On the Ixia Tx port create two flows - a) 'Test Data Traffic' and
+       b) 'Background Data traffic'.
+    4. The flow 'Test Data Traffic' can assume one of the lossless priority 
+       values Pi.
+    5. The flow 'Background Data Traffic' can assume all the priority values
+       which are not in 'Test Data Traffic'. For example if the priority of
+       'Test Data Traffic' is 3, the priorities of the 'Background Data 
+       Traffic' should be 0, 1, 2, 4, 5, 6, 7.   
+    6. From Rx port send pause frames on priority Pi. Such that priority of
+       'Test Data Traffic' at Tx end == Pause Priority at Rx end. That is,
+       send pause frames on priority Pi.
+    7. Start 'Test Data Traffic' and 'Background Data Traffic'.
+    8. You may repeat step 6 and 7 for each lossless priorities.
+    9. Expected result -
+       a. No 'Test Data Traffic' will flow. Since priority of
+          'Test Data Traffic' equals to the priority of PFC pause frames.
+       b. 'Background Data Traffic' will always flow.
+
+    Note: Test and background traffic should be started after PFC pause storm.
+
+    """ 
+    port_list = list()
+    fanout_devices = IxiaFanoutManager(fanout_graph_facts)
+    fanout_devices.get_fanout_device_details(device_number=0)
+    device_conn = conn_graph_facts['device_conn']
+
+    for intf in fanout_devices.get_ports():
+        peer_port = intf['peer_port']
+        intf['speed'] = int(device_conn[peer_port]['speed'])
+        port_list.append(intf)
+
+    # The topology should have at least two interfaces.
+    pytest_assert(len(device_conn) >= 2,
+        "The topology should have at least two interfaces")
+
+    # Test pausing each lossless priority individually.
+    session = ixia_api_server_session
+    for prio in lossless_prio_dscp_map:
+        send_pause_frame = False
```

Comment:
Add another loop: for send_pause_frame in [True, False]:

Reply:
Done